### PR TITLE
Audit links

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Updated base styling of links
- Update `.external` link pattern to become `.p-link--external`
- Update `.link-top` link pattern to become `.p-link--top`
- Update documentation
- Temporarily add back `demo` folder for QA purposes

## QA

- Sanity check code
- Run `gulp develop` and check `demo/index.html` for new and improved links as above.

## Details

Fixes: #417

### CODE REVIEW

File: scss/_base_links.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Base styling for links
+@mixin vf-b-links {
+  a {
```
Comment (anthonydillon) [inline scss/_base_links.scss]: No hover styling?

File: docs/patterns/_buttons.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ title: buttons
 
 <div class="row">
     <div class="six-col">
-        <button class="button--primary" href="#"><span class="external">External button</span></button>
+        <button class="button--primary" href="#"><span class="link--external">External button</span></button>
```
Comment (anthonydillon) [inline docs/patterns/_buttons.md]: Thought it was `p-link- external`

File: docs/patterns/_buttons.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ title: buttons
 
 <div class="row">
     <div class="six-col">
-        <a class="button--secondary" href="#"><span class="external">External secondary button</span></a>
+        <a class="button--secondary" href="#"><span class="link--external">External secondary button</span></a>
```
Comment (anthonydillon) [inline docs/patterns/_buttons.md]: Again here

File: docs/patterns/_grid-list.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ title: grid list
                 <img class="grid-list__img" src="https://placeholdit.imgix.net/~text?txtsize=8&amp;txt=60%C3%9760&amp;w=60&amp;h=60" alt="icon">
             </div>
             <div class="three-col last-col">
-                <h3><a class="external" href="#">Link</a></h3>
+                <h3><a class="link--external" href="#">Link</a></h3>
```
Comment (anthonydillon) [inline docs/patterns/_grid-list.md]: `p-link-external`

File: docs/patterns/_grid-list.md [hunk 5]
```diff
@@ -20,7 +20,7 @@ title: grid list
...
```
Comment (anthonydillon) [inline docs/patterns/_grid-list.md]: Again

Comment (anthonydillon) [inline docs/patterns/_grid-list.md]: Here too

File: docs/patterns/_grid-list.md [hunk 6]
```diff
@@ -38,7 +38,7 @@ title: grid list
...
```
Comment (anthonydillon) [inline docs/patterns/_grid-list.md]: Here

**Comment by anthonydillon** [comment]
Thanks. Seems with a few comments. Could you briefly explain why these are patterns and not utility classes? Seems to me more utility based but might be missing something.

File: scss/_base_links.scss (hunk 1 above)
Comment (barrymcgee) [inline scss/_base_links.scss]: No :hover or :visited styling has been provided so far and I've already raised this with @yaili as a separate issue in #448

File: docs/patterns/_buttons.md (hunk 2 above)
Comment (barrymcgee) [inline docs/patterns/_buttons.md]: Hmm.. seems like I botched the last rebase, sorry - fixing up now.

**Comment by barrymcgee** [comment]
@anthonydillon Placing styling within any one layer over another is always subjective to a certain degree but I feel utilities and helpers should be used as overrides, in ITCSS terminology, "trumping" what has come before - Harry Roberts recommends adding `!important` to all rules in this layer for this reason.

Although also a slightly subjective directive, I also feel a utility selector should contain relatively few rules, maybe only one or two to ensure it's very specific to what it describes. `.u-float-right {float: right!important}` whereas a more complex set of rules such as these probably signifies a pattern.

**Comment by anthonydillon** [comment]
Agree its subjective. Just thought I should raise it. :+1:

**Pull request merged**